### PR TITLE
[Hub Generated] Review request for Microsoft.HybridCompute to add version preview/2021-05-17-preview

### PR DESCRIPTION
**Pull request by dkirby017**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/dkirby017/azure-rest-api-specs/dev-hybridcompute-Microsoft.HybridCompute-2021-05-17-preview...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-05-17-preview/privateLinkScopes.json [hunk 1]
```diff
@@ -800,6 +800,14 @@
           "readOnly": true,
           "description": "The Guid id of the private link scope.",
           "type": "string"
+        },
+        "privateEndpointConnections": {
```
Comment (RamyasreeChakka) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-05-17-preview/privateLinkScopes.json:804]: I see privateEndpointConnections array defined here contains the list of connections and the details. Few questions...

1. How many privateEndpointConnections can exist? Would all the connections details fit in a single response? Note that you cannot do paging just for the array content. 
2. Also you will run into RBAC issues if you return the entire privateEndpointConnection response in privateLinkScope GET. Refer https://armwiki.azurewebsites.net/api_contracts/guidelines/rpc.html#rpc019-rbac-violation-information-leak-when-resources-full-content-of-other-types-are-included-in-the-response

I would recommend to return just an array of privateEndpointConnection IDs.

Comment (dkirby017) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-05-17-preview/privateLinkScopes.json:804]: 1. Single digits. Most likely 1, but we wanted to make sure we had support built in for >1
2. The privateEndpointConnection data isn't really a different resource type, but just data on a privateLinkScope. it also isn't something that should have different RBAC rules associated with it. This information is needed for policy checks, so we need it to be available here.

Comment (dkirby017) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-05-17-preview/privateLinkScopes.json:804]: adding on to 2., the data we're returning here isn't the underlying private endpoint resource itself either. It's just some information about the private link scope's relationship with that private endpoint, plus the id of the private endpoint. I don't think there should be any information leak issues here.

Comment (RamyasreeChakka) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2021-05-17-preview/privateLinkScopes.json:804]: @dkirby017 Sounds good then. Signing off by adding 'ARMSignedOff' label.

**Review by RamyasreeChakka — APPROVED** [review]
:shipit:

**Pull request merged**